DEV 83 1/2: Update Footer to the new Design

ApplePieAngel opened the pull request:
**Merge notes: Will need to be merged with #73 at least. (Can ignore AppBar)**

# Description
The purpose of this change is to update the footer component to align with the new design created by the design team.

Reasons for changes: 
- Non external links have been changed to useHistory in order to maintain userHistory when navigating pages
- Grid and CSS has been added to centralize the font/make the footer more responsive
- Donate doesn't actually redirect to any page yet, but was added as its displayed in the Figma design.
**(12/28/2022 Update)**
I've added unit test cases covering the links and copyright string to this PR.
- MUI palette colors doesn't play nice with unit tests. They always render as undefined and error out the tests. So I've converted it back to hardcoded values.

The reasoning behind my test cases styling is inspired from "Mastering React Test-Driven Development" by Daniel Irvine.

Related to DEV-83

## Screenshots
**Before**

![chrome_2bGK1lIYdu](https://user-images.githubusercontent.com/86702974/200721520-bcc00231-65cf-42ce-bcac-ee72534f718d.png)

![chrome_3P8anlP15J](https://user-images.githubusercontent.com/86702974/200721536-279545bd-1542-4892-a649-55c64d6be888.png)

**After**

![chrome_9tXUwiP911](https://user-images.githubusercontent.com/86702974/200721507-c1f35255-7fad-44ba-b341-fab625dc0be6.png)

![chrome_vSeCdDjXyc](https://user-images.githubusercontent.com/86702974/200721528-b348f40c-c4d4-4423-be08-322494d07dd2.png)


## Type of change
Design Change

# How Has This Been Tested?
Using Chrome Browser + Chrome Developer Tools to test various breakpoints. 
As well as manually clicking through links. 
**Added and tested with unit tests to cover the components.**

deepakvenkat commented on src/components/Footer/Footer.jsx:
```diff
@@ -26,88 +30,96 @@ const styles = (theme) => ({
+          <Button
+            className={classes.buttonLink}
+            onClick={() => {
+              history.push({
```
Why do we need this instead of just a `href` ?

ApplePieAngel replied:
Href elements would trigger a page refresh which would reset the application states. In this instance it means that the application would forget if the user was logged in or not which is relevant for other components and single page applications. 

If not useHistory, we can also use NavLink or Link which is part of React Router.

ApplePieAngel commented:
As an update, I've updated all the Buttons to NavLinks. I've also added underlines under external links as I felt it wasn't clear to users if they were links previously.

![Code_wlVp4SOy0w](https://user-images.githubusercontent.com/86702974/201427051-011f547e-bc66-473c-9c8e-b081def6fa26.png)
![Code_afeQj0rZzF](https://user-images.githubusercontent.com/86702974/201427068-36634bef-d645-4765-ac2b-75a19d985b2f.png)

ApplePieAngel commented:
As another update, I know several months ago we talked about not liking how testing was currently done. I've done a lot of research (books, conference talks, articles, etc) and added unit tests for this PR as my recommendation and my proposal for how unit tests should look like going forward.

InManuBytes reviewed (approved):
LGTM
